### ISSUE TITLE
SQL map functions not working anymore

### ISSUE THREAD

**Issue by kevmegforest** [orientechnologies/orientdb#8714]
### OS:  Docker 

This is the command I use
```
select calc.@class as calc_class, map(diag, diag.Flag) as diags  FROM (match {class: diagnostics, as: diag, where : (Store = 'IGA_Auteuil' and End_Date = '2018-10-01 00:00:00' and Timespan in ['Monthly','Weekly','Daily']) }.out(EHasDiagnostics) {as: calc} return calc , diag ) group by calc.@class
```

## Expected behavior  
In orientdb 2.2.37, the diag records where grouped in a new map of the categories

like this 

```
{
    "result": [
        {
            "@type": "d",
            "@rid": "#-2:0",
            "@version": 0,
            "calc_class": "VHotte",
            "diags": {
                "#185:1469": "Yellow",
                "#187:1042": "Red",
                "#188:1042": "Green",
                "#187:619": "Green",
                "#187:190": "Red",
                "#188:1468": "Red"
            }
        },
        {
            "@type": "d",
            "@rid": "#-2:2",
            "@version": 0,
            "calc_class": "VLighting",
            "diags": {
                "#228:1101": "Green",
                "#226:1547": "Green",
                "#227:1101": "Green",
                "#225:205": "Green",
                "#227:654": "Green",
                "#228:654": "Green"
            }
        },
        {
            "@type": "d",
            "@rid": "#-2:6",
            "@version": 0,
            "calc_class": "VWattNode_Total",
            "diags": {
                "#230:10": "Green"
            }
        },
        {
            "@type": "d",
            "@rid": "#-2:7",
            "@version": 0,
            "calc_class": "VAntisweat",
            "diags": {
                "#221:247": "Green"
            }
        },
        {
            "@type": "d",
            "@rid": "#-2:9",
            "@version": 0,
            "calc_class": "VDeshumStage",
            "diags": {
                "#168:91": "Yellow",
                "#168:306": "Yellow"
            }
        },
        {
            "@type": "d",
            "@rid": "#-2:3",
            "@version": 0,
            "calc_class": "VCOP",
            "diags": {
                "#199:63": "Green",
                "#198:534": "Green",
                "#189:599": "Green",
                "#194:82": "Green"
            }
        },
        {
            "@type": "d",
            "@rid": "#-2:8",
            "@version": 0,
            "calc_class": "VSpaceTemp",
            "diags": {
                "#153:923": "Green",
                "#156:540": "Green"
            }
        },
        {
            "@type": "d",
            "@rid": "#-2:5",
            "@version": 0,
            "calc_class": "VDeshumDXCoils",
            "diags": {
                "#169:47": "Yellow"
            }
        },
        {
            "@type": "d",
            "@rid": "#-2:1",
            "@version": 0,
            "calc_class": "VCompressor",
            "diags": {
                "#220:3141": "Green",
                "#220:4443": "Green",
                "#220:1843": "Green",
                "#217:4444": "Green",
                "#218:3141": "Green",
                "#219:3141": "Green",
                "#218:4444": "Green",
                "#219:4444": "Green",
                "#217:3141": "Green",
                "#220:541": "Green",
                "#216:714": "Green",
                "#217:1844": "Green",
                "#219:541": "Green",
                "#218:541": "Green",
                "#218:1844": "Green",
                "#219:1844": "Green",
                "#217:3142": "Green",
                "#213:422": "Green",
                "#217:542": "Green"
            }
        },
        {
            "@type": "d",
            "@rid": "#-2:4",
            "@version": 0,
            "calc_class": "VPreheat",
            "diags": {
                "#163:101": "Green",
                "#164:841": "Green"
            }
        }
    ],
    "notification": "Query executed in 0.073 sec. Returned 10 record(s)"
}
```

## Actual behavior  
But in orientdb  3.0.11, the results are not grouped 

```
{
    "result": [
        {
            "calc_class": "VCompressor",
            "diags": {
                "#218:1435": "Green"
            }
        },
        {
            "calc_class": "VCompressor",
            "diags": {
                "#217:461": "Green"
            }
        },
        {
            "calc_class": "VHotte",
            "diags": {
                "#187:1540": "Green"
            }
        },
        {
            "calc_class": "VCompressor",
            "diags": {
                "#215:1060": "Green"
            }
        },
        {
            "calc_class": "VSpaceTemp",
            "diags": {
                "#154:1271": "Green"
            }
        },
        {
            "calc_class": "VPreheat",
            "diags": {
                "#164:518": "Green"
            }
        },
        {
            "calc_class": "VCompressor",
            "diags": {
                "#215:105": "Green"
            }
        },
        {
            "calc_class": "VCompressor",
            "diags": {
                "#217:1111": "Green"
            }
        },

...

```

**Comment by kevmegforest** [comment]
I made some changes to my function and used a list instead of map and it worked.
So the only thing not working is in fact the map function. 

Just with the demodb  doing the following commands should be pretty clear:
```
Select  list(@rid) from V 
Select  map(@rid, @class) from V 
```
The list one gives one record (aggregated)
The map one gives one map per record in V instead of only one record consisting of the aggregation

It does not aggregate like the documentation says it should :

map()
Adds a value to a map. The first time the map is created. If <value> is a map, then is merged with the map, otherwise the pair <key> and <value> is added to the map as new entry.
Syntax: map(<key>, <value>)
Available since: 1.2.0

**Comment by luigidellaquila** [comment]
Hi @kevmegforest 

I'm checking it now, I'll let you know ASAP

Thanks

Luigi

**Comment by luigidellaquila** [comment]
Hi @kevmegforest 

I just pushed a fix for this problem, it will be in the snapshot in a few minutes and will be released with v 3.0.14

Thanks

Luigi
